medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -247,6 +278,193 @@ mod test {
         );
     }
 
+    #[test]
+    fn failure_to_delete_blob_leaves_job_status_as_accepted() {
+        let mut state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_a_delete_blob_error();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            state_manager.get_most_recently_set_status(),
+            JobStatus::Accepted
+        );
+    }
+
+    #[test]
+    fn failure_to_delete_blob_does_not_remove_message_from_service_bus() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_a_delete_blob_error();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            removeable_message.remove_was_called, false,
+            "Removed message, but shouldn't"
+        );
+    }
+
+    #[test]
+    fn failure_to_restore_index_removes_message_since_cannot_be_retried() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_failure_to_restore_index();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            removeable_message.remove_was_called, true,
+            "Didn't remove message, but should"
+        );
+    }
+
+    #[test]
+    fn failure_to_restore_index_leaves_job_status_as_error() {
+        let mut state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_failure_to_restore_index();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            state_manager.get_most_recently_set_status(),
+            JobStatus::Error {
+                message: String::from("Cannot restore index for blob Blob Id: Error message"),
```

Comment:
not immediately clear where `"Blob Id"` is coming from – might be worth passing a generated id into `given_failure_to_restore_index`

Reply:
True - added